feat(defaults): store spellfile in stdpath('data') (#33048)
neovim/neovim: runtime/doc/news.txt

These existing features changed their behavior.

> • |spellfile| uses `stdpath('data')/spell` as default location instead of the

Reviewer: ```suggestion
• |spellfile| location defaults to `stdpath('data').."/spell"` instead of the
```
